dialogue-core/src/main/java/com/palantir/dialogue/core/AimdConcurrencyLimiter.java

```diff
@@ -23,7 +23,11 @@
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.IntBinaryOperator;
 
-/** Simple lock-free additive increase multiplicative decrease concurrency limiter. */
+/**
+ * Simple lock-free additive increase multiplicative decrease concurrency limiter. Typically, a dispatching
+ * {@link com.palantir.dialogue.Request} tries to {@link #acquire} a new token and releases it when the
+ * corresponding {@link Response} is retrieved.
```

Comment:
Some javadoc is a great idea here! Maybe worth referring to netflix's AIMD limit as the inspiration, and that this one just avoids needing `synchronized` blocks?

Reply:
This avoids a lot beyond synchronization, reading from the precise system clock for example. Their library is great, but has a lot of code to support extension and configurability in ways we don’t need.

Reply:
added link